tests/common.py

```diff
@@ -1050,3 +1051,85 @@ def mock_signal_handler(*args):
     hass.helpers.dispatcher.async_dispatcher_connect(signal, mock_signal_handler)
 
     return calls
+
+
+class hashdict(dict):
+    """
+    hashable dict implementation, suitable for use as a key into other dicts.
+
+        >>> h1 = hashdict({"apples": 1, "bananas":2})
+        >>> h2 = hashdict({"bananas": 3, "mangoes": 5})
+        >>> h1+h2
+        hashdict(apples=1, bananas=3, mangoes=5)
+        >>> d1 = {}
+        >>> d1[h1] = "salad"
+        >>> d1[h1]
+        'salad'
+        >>> d1[h2]
+        Traceback (most recent call last):
+        ...
+        KeyError: hashdict(bananas=3, mangoes=5)
+
+    based on answers from
+       http://stackoverflow.com/questions/1151658/python-hashable-dicts
+
+    """
+
+    def __key(self):  # noqa: D105 no docstring
+        return tuple(sorted(self.items()))
+
+    def __repr__(self):  # noqa: D105 no docstring
+        return ", ".join("{0}={1}".format(str(i[0]), repr(i[1])) for i in self.__key())
+
+    def __hash__(self):  # noqa: D105 no docstring
+        return hash(self.__key())
+
+    def __setitem__(self, key, value):  # noqa: D105 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def __delitem__(self, key):  # noqa: D105 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def clear(self):  # noqa: D102 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def pop(self, *args, **kwargs):  # noqa: D102 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def popitem(self, *args, **kwargs):  # noqa: D102 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def setdefault(self, *args, **kwargs):  # noqa: D102 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    def update(self, *args, **kwargs):  # noqa: D102 no docstring
+        raise TypeError(
+            "{0} does not support item assignment".format(self.__class__.__name__)
+        )
+
+    # update is not ok because it mutates the object
+    # __add__ is ok because it creates a new object
+    # while the new object is under construction, it's ok to mutate it
+    def __add__(self, right):  # noqa: D105 no docstring
+        result = hashdict(self)
+        dict.update(result, right)
+        return result
+
+
+def assert_lists_same(a, b):
+    """Compare two lists, ignoring order."""
+    assert collections.Counter([hashdict(i) for i in a]) == collections.Counter(
+        [hashdict(i) for i in b]
+    )
```

Comment:
Added to compare lists with different order since the order of automations is not deterministic when we have automations from different domains.